data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/AbstractCosmosOperations.java

```diff
@@ -0,0 +1,546 @@
+import io.micronaut.data.annotation.Query;
+import io.micronaut.data.cosmos.common.Constants;
+import io.micronaut.data.cosmos.common.CosmosDatabaseInitializer;
+import io.micronaut.data.cosmos.config.CosmosDatabaseConfiguration;
+import io.micronaut.data.exceptions.DataAccessException;
+import io.micronaut.data.intercept.annotation.DataMethod;
+import io.micronaut.data.model.DataType;
+import io.micronaut.data.model.PersistentEntity;
+import io.micronaut.data.model.PersistentProperty;
+import io.micronaut.data.model.PersistentPropertyPath;
+import io.micronaut.data.model.query.builder.sql.SqlQueryBuilder;
+import io.micronaut.data.model.runtime.AttributeConverterRegistry;
+import io.micronaut.data.model.runtime.PreparedQuery;
+import io.micronaut.data.model.runtime.QueryParameterBinding;
+import io.micronaut.data.model.runtime.RuntimeEntityRegistry;
+import io.micronaut.data.model.runtime.RuntimePersistentEntity;
+import io.micronaut.data.model.runtime.RuntimePersistentProperty;
+import io.micronaut.data.model.runtime.StoredQuery;
+import io.micronaut.data.model.runtime.convert.AttributeConverter;
+import io.micronaut.data.runtime.config.DataSettings;
+import io.micronaut.data.runtime.convert.DataConversionService;
+import io.micronaut.data.runtime.date.DateTimeProvider;
+import io.micronaut.data.runtime.operations.internal.AbstractRepositoryOperations;
+import io.micronaut.data.runtime.operations.internal.sql.DefaultSqlPreparedQuery;
+import io.micronaut.data.runtime.operations.internal.sql.DefaultSqlStoredQuery;
+import io.micronaut.data.runtime.operations.internal.sql.SqlPreparedQuery;
+import io.micronaut.data.runtime.operations.internal.sql.SqlStoredQuery;
+import io.micronaut.data.runtime.query.MethodContextAwareStoredQueryDecorator;
+import io.micronaut.data.runtime.query.PreparedQueryDecorator;
+import io.micronaut.http.codec.MediaTypeCodec;
+import io.micronaut.jackson.core.tree.JsonNodeTreeCodec;
+import io.micronaut.json.tree.JsonNode;
+import io.micronaut.serde.Decoder;
+import io.micronaut.serde.Deserializer;
+import io.micronaut.serde.SerdeRegistry;
+import io.micronaut.serde.Serializer;
+import io.micronaut.serde.jackson.JacksonDecoder;
+import io.micronaut.serde.support.util.JsonNodeEncoder;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.Time;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+/**
+ * The base class for Cosmos Db operations.
+ *
+ * @author radovanradic
+ * @since 4.0.0
+ */
+abstract class AbstractCosmosOperations extends AbstractRepositoryOperations implements
```

Comment:
We can probably remove the abstract class

Reply:
Ugh, yes, I agree. It's just I created new branch based on this so will be difficult to merge:) But I will remove it, makes sense.